### PR TITLE
docs: Corrections/extra information in design.rst

### LINKED ISSUE #3725: Mode used in Node/Tree is specific to go and might change in future go versions?
Sorry for not using the template. This issue is not a concrete bug, but something I observed when developing [rustic](https://github.com/rustic-rs/rustic)

The point is that in `internal/restic/node.go` the following mask is used for the modes which are saved in the trees:
```
mask := os.ModePerm | os.ModeType | os.ModeSetuid | os.ModeSetgid | os.ModeSticky
```

The [docu](https://pkg.go.dev/io/fs#ModeType) says:
> The defined file mode bits are the most significant bits of the FileMode. The nine least-significant bits are the standard Unix rwxrwxrwx permissions. The values of these bits should be considered part of the public API and may be used in wire protocols or disk representations: they must not be changed, although new bits might be added. 

With other words: restic uses part of the non-public API and saves this information in the on-disc representation.

For rustic, this has the effect that I wasn't able to easily reproduce the file modes (even though the last nine bits were always correct). The effect is that when using rustic on an restic repository and backup an completely unchanged path, it correctly detects the parent snapshot and also correctly identifies that no file has been changed - but most or all trees are newly written as the modes changed. Hence, afterwards, two versions of the trees are saved which means double storage used for trees.

I don't know how stable this non-public API in go is, but if it will happen to change, users will experience the same when using restic compiled with different go versions on the same repository.

Actually I don't know a good way to solve this. My impulse would be to only save the mode umasked by `0777`, but this will simply enforce the unwanted behavior I depicted above right now...

### PR DESCRIPTION
**Pull request by aawsome**
What does this PR change? What problem does it solve?
-----------------------------------------------------

Updates the design document. Corrects some wrong/misleading parts and adds information about tree meta data.

Was the change previously discussed in an issue or on the forum?
----------------------------------------------------------------

closes #3725
see also #3808

Checklist
---------

- [x] I have read the [contribution guidelines](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#providing-patches).
- [x] I have [enabled maintainer edits](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [N/A] I have added tests for all code changes.
- [N/A] I have added documentation for relevant changes (in the manual).
- [ ] There's a new file in `changelog/unreleased/` that describes the changes for our users (see [template](https://github.com/restic/restic/blob/master/changelog/TEMPLATE)) - don't think that's worth it. Feel free to add one!
- [N/A] I have run `gofmt` on the code in all commits.
- [x] All commit messages are formatted in the same style as [the other commits in the repo](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#git-commits).
- [x] I'm done! This pull request is ready for review.

### CODE REVIEW

File: doc/design.rst [hunk 1]
```diff
@@ -45,10 +45,12 @@ comparing its output to the file name. If the prefix of a filename is
...
-Apart from the files stored within the ``keys`` directory, all files are
-encrypted with AES-256 in counter mode (CTR). The integrity of the
-encrypted data is secured by a Poly1305-AES message authentication code
-(sometimes also referred to as a "signature").
+Apart from the files stored within the ``keys`` and ``data``directories,
+all files are encrypted with AES-256 in counter mode (CTR). The integrity
+of the encrypted data is secured by a Poly1305-AES message authentication
+code (sometimes also referred to as a "signature").
+Files in the ``data`` directory ("pack files") consist of multiple parts
+which are all independently enccrypted and authenticated, see below.
```
Comment (greatroar) [inline doc/design.rst:53]:
```suggestion
Apart from the files stored within the ``keys`` and ``data`` directories,
all files are encrypted with AES-256 in counter mode (CTR). The integrity
of the encrypted data is secured by a Poly1305-AES message authentication
code (sometimes also referred to as a "signature").
Files in the ``data`` directory ("pack files") consist of multiple parts
which are all independently encrypted and authenticated, see below.
```

File: doc/design.rst [hunk 2]
```diff
@@ -298,8 +300,8 @@ example, the Pack ``73d04e61`` contains two data Blobs and one Tree
 blob, the plaintext hashes are listed afterwards. The ``length`` field
 corresponds to ``Length(encrypted_blob)`` in the pack file header.
 Field ``uncompressed_length`` is only present for compressed blobs and
-therefore is never present in version 1. It is set to the value of
-``Length(blob)``.
+therefore is never present in the Repository format version 1. It is set
```
Comment (rawtaz) [inline doc/design.rst:303]: Total nitpick but wouldn't ".. never present in version 1 of the repository format" be better wording? (Not sure why there would be a capital R in "repository" either, it's not in the earlier text).

File: doc/design.rst [hunk 3]
```diff
@@ -495,9 +501,18 @@ the JSON is indented):
...
 A tree contains a list of entries (in the field ``nodes``) which contain
-meta data like a name and timestamps. When the entry references a
-directory, the field ``subtree`` contains the plain text ID of another
-tree object.
+meta data like a name and timestamps. Note that there are some specialities of how
+this metadata is generated:
+
+- The name is quoted using <https://pkg.go.dev/strconv#Quote> before being saved.
+  This handles non-unicode names, but also changes the representation of names
+  containing `"` or `\\`.
```
Comment (rawtaz) [inline doc/design.rst:509]: Should probably be dual backticks on each side of the words.

**Review by MichaelEischer — APPROVED** [review]
LGTM except for the nitpicks already mentioned by rawtaz.

**Comment by MichaelEischer** [comment]
I've addressed the review comments and improve the link formatting.

**Pull request merged**